### DISCUSSION TITLE
Suggestion: Add IGDB endpoints

### DISCUSSION THREAD

**Discussion by Katzen48** [romanzipp/Laravel-Twitch#84] [Ideas]
Twitch switched their game lookup from Giant Bomb to IGDB (which they bought a while ago).
See https://twitter.com/TwitchSupport/status/1310625492274470912
With that switch, IGDB's API now uses Twitch' OAuth2 Tokens (https://medium.com/igdb/igdb-api-v4-is-coming-6ba97874edbc).

I think, it would be nice, if IGDB (which is now technically part of Twitch) could be integrated into this package.
Generating and handling Bearer Tokens for the same authentication system twice, seems a like a big overhead.

Additionally, by integrating the IGDB API, more information regarding the games, returned by Twitch's API, can be obtained without handling another authentication system.

**Comment by ghostzero** [comment]
For me IGDB is a completely decoupled product that has a completely different namespace. The only thing they share is the oauth server.

Also a quote from d-fischer / SatisfiedPear from Twitch API Discord:
> Same stance as on my own package: IGDB should be a separate package, it could maybe depend on the auth portion of the Twitch code but it has nothing else in common, really

**Comment by Katzen48** [comment]
Still, Twitch's game library is now powered by IGDB.
To be able to request additional data from the same data source using another package and another token handling seems a bit weird.
It doesn't have to be directly in this package, but a possibility to request metadata is required.

> The only thing they share is the oauth server.

Nope, if you look at https://medium.com/igdb/we-are-joining-twitch-54cfdf44c496 they moved them to the same infrastructure. It's all Twitch. (see "By joining Twitch, we will be able to tap into their experience, **resources**," or "clean up some features and **migrate infrastructure** in order to make room for new and improved tools.")

Twitch changes the API and moves it to the same infrastructure, to make it more accessible for Twitch-related developers.
So there is no reason to not allow PHP developers, who want to build something Twitch-related, to use one system for all Twitch APIs instead of a dozen (one for every namespace).

At least, the app access token should be made available to outside packages, to be able to use the same token and not requiring to request another one.
This way, another package could hook into this library and allow getting more information about the game returned by the Kraken/Helix API.
